src/layers/core/hexagon-layer/hexagon-layer.js

```diff
@@ -94,17 +127,35 @@ export default class HexagonLayer extends Layer {
     });
   }
 
+  _onPercentileChange() {
+    const {lowerPercentile, upperPercentile} = this.props;
+
+    this.state.valueDomain = this.state.sortedCounts
+      .getCountRange([lowerPercentile, upperPercentile]);
+  }
+
   _onGetSublayerColor(cell) {
     const {colorRange} = this.props;
-    const colorDomain = this.props.colorDomain || this.state.countRange;
+    const {valueDomain} = this.state;
+
+    const colorDomain = this.props.colorDomain || valueDomain;
+    const count = cell.points.length;
+    const color = quantizeScale(colorDomain, colorRange, count);
```

Comment:
`quantizeScale` is implemented from scratch in `utils/scale-util`. We are not depend on d3 for this